crates/bevy_ecs/src/lib.rs

```diff
@@ -1793,6 +1793,78 @@ mod tests {
         );
     }
 
+    #[test]
+    fn required_components() {
+        #[derive(Component)]
+        #[require(Y)]
+        struct X;
+
+        #[derive(Component)]
+        #[require(Z(new_z))]
+        struct Y {
+            value: String,
+        }
+
+        #[derive(Component)]
+        struct Z(u32);
+
+        impl Default for Y {
+            fn default() -> Self {
+                Self {
+                    value: "hello".to_string(),
+                }
+            }
+        }
+
+        fn new_z() -> Z {
```

Comment:
This would be clearer as `construct_z` or something: `new_z` read as a setter to me.

Reply:
I would prefer to not use "construct" here, given that that term will mean something specific soon, and it will relate to this code.

Reply:
I don't see how `new_z` would read as a setter. I can't think of any cases where I've seen that convention. I've seen plenty of cases where `new_x` initializes a new X though.